src/Microsoft.Dnx.Tooling/Publish/PublishManager.cs

```diff
@@ -57,8 +56,18 @@ public bool Publish()
                 _options.Reports.Information.WriteLine(warning.FormattedMessage.Yellow());
             }
 
-            // '--wwwroot' option can override 'webroot' property in project.json
-            _options.WwwRoot = _options.WwwRoot ?? project.WebRoot;
+            if (File.Exists(Path.Combine(project.ProjectDirectory, "Microsoft.AspNet.Hosting.json")))
+            {
+                var jsonObj = JObject.Parse(File.ReadAllText(Path.Combine(project.ProjectDirectory, "Microsoft.AspNet.Hosting.json")));
+                _options.WwwRoot = _options.WwwRoot ?? jsonObj?["Hosting"]?["WebRoot"]?.ToString();
```

Comment:
why do all the parsing if you won't use that if `_options.WwwRoot` is not null? Should the `if` condition be changed to: `if (_options.WwwRoot == null && File.Exists(Path.Combine(project.ProjectDirectory, "Microsoft.AspNet.Hosting.json")))`. Same below.